File: src/src_core/fmg/core/types/draw/EZoomInterface.cs
using System;

namespace Fmg.Core.Types.Draw {

    public enum EZoomInterface {
        eAlwaysMax,
        eMax, eMin,
        eInc, eDec
    }

    public static class EZoomInterfaceExt {
        public static string getDescription(this EZoomInterface self) {
            switch (self) {
            case EZoomInterface.eAlwaysMax: return "Always maximal size";
            case EZoomInterface.eMax: return "Maximal mosaic size";
            case EZoomInterface.eMin: return "Minimal mosaic size";
            case EZoomInterface.eInc: return "Increase mosaic size";
            case EZoomInterface.eDec: return "Decrease mosaic size";
            }
            throw new Exception("Invalid " + self.ToString());
        }
    }

}

File: FastMines_swing/src/main/java/fmg/swing/app/menu/EZoomInterface.java
package fmg.swing.app.menu;

public enum EZoomInterface {

    eAlwaysMax,
    eMax, eMin,
    eInc, eDec;

    public String getDescription() {
        switch (this) {
        case eAlwaysMax: return "Always maximal size";
        case eMax      : return "Maximal mosaic size";
        case eMin      : return "Minimal mosaic size";
        case eInc      : return "Increase mosaic size";
        case eDec      : return "Decrease mosaic size";
        }
        throw new RuntimeException("Invalid "+this.toString());
    }

}
